erpnext/projects/doctype/project/project.json

```diff
@@ -438,13 +439,15 @@
    "description": "Message will sent to users to get their status on the project",
    "fieldname": "message",
    "fieldtype": "Text",
-   "label": "Message"
+   "label": "Message",
+   "mandatory_depends_on": "collect_progress"
```

Comment:
@marination v12 doesn't have `mandatory_depends_on` feature. You'll have to use `set_df_property` and set `reqd` if `collect_progress` is checked.

Reply:
@hrwX  Done